shared/go/aws/cloudwatchlogs/cloudwatch.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright (C) 2022 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloudwatchlogs
+
+import (
+	"context"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudwatchlogs"
+	"github.com/falcosecurity/plugins/shared/go/aws/session"
+)
+
+const (
+	defaultShift      int64 = 1   // controls the time shift in past for the first call to CloudwatchLogs API
+	defaultRefresh    int64 = 5   // time between two calls to CloudwatchLogs API
+	defaultBufferSize int64 = 200 // buffer size of the channel that transmits Logs to the Plugin
+)
+
+// Filter represents a filter for retrieving logs from CloudwatchLogs API
+type Filter struct {
+	FilterPattern       string
+	LogGroupName        string
+	LogStreamNamePrefix string
+	LogStreamNames      string
+}
+
+// Client represents a client for CloudwatchLogs API
+type Client struct {
+	*cloudwatchlogs.CloudWatchLogs
+}
+
+// Options represents options for calls to CloudwatchLogs API
+type Options struct {
+	Shift      int64
+	Refresh    int64
+	BufferSize int64
+}
+
+// CreateFilter returns a Client for retrieving logs from CloudwatchLogs API
+func CreateFilter(filterPattern, logGroupName, logStreamNamePrefix, logStreamNames string) *Filter {
+	return &Filter{
+		FilterPattern:       filterPattern,
+		LogGroupName:        logGroupName,
+		LogStreamNamePrefix: logStreamNamePrefix,
+		LogStreamNames:      logStreamNames,
+	}
+}
+
+// CreateFilter returns a Filter for CloudwatchLogs API
+func CreateClient(cfgs *aws.Config) *Client {
+	return &Client{
+		CloudWatchLogs: cloudwatchlogs.New(session.CreateSession(), cfgs),
+	}
+}
+
+// setDefault set the default values for Options
+func (options *Options) setDefault() {
+	if options.Refresh <= 1 {
+		options.Refresh = defaultRefresh
+	}
+	if options.Shift < 0 {
+		options.Refresh = defaultShift
+	}
+	if options.BufferSize <= 0 {
+		options.Refresh = defaultBufferSize
+	}
```

Comment:
```suggestion
       if options.Refresh == 0 {
		options.Refresh = defaultRefresh
	}
	if options.Shift == 0 {
		options.Shift = defaultShift
	}
	if options.BufferSize == 0 {
		options.BufferSize = defaultBufferSize
	}
```
Note that you're just assigning options.Refresh here. Also, changing the option types as suggested above allows you to just check for their 0 values